designs/node-repair.md

```diff
@@ -0,0 +1,93 @@
+
+Karpenter users are observing that occasionally nodes can face failure mode that prevent them to be healthy in a Kubernetes clusters. These nodes stay in unhealthy state for a period of time without any actions, and in some cases forever with manual intervention. Karpenter continues to simulate pods to these nodes, but the Kubernetes scheduler won't schedule to them. This will leave pods unscheduled and cause downtime for applications. Customers have the need for karpenter to repair unhealthy nodes. The karpenter team has seen several failure mode such as disk pressure, memory pressure, networking outage etc. that can cause nodes to get into an unhealthy state Here are issues related to the topic:
+
+* Mega Issue: https://github.com/kubernetes-sigs/karpenter/issues/750
+    * Related (Unreachable): https://github.com/aws/karpenter-provider-aws/issues/2570
+    * Related (Remove by taints): https://github.com/aws/karpenter-provider-aws/issues/2544
+    * Related (Known resource are not registered) Fixed by v0.28.0: https://github.com/aws/karpenter-provider-aws/issues/3794
+    * Related (Stuck on NotReady): https://github.com/aws/karpenter-provider-aws/issues/2439
+
+#### Out of scope
+
+The scope of the solution suggested below will be an opinionated method that will not give customer configurability. The team does not have enough data to determine the right level of configuration surface that users will utilize. The implementation will not consider budgets, API surface for repairing unhealthy nodes, or option for nodes to be rebooted instead of replaced. The solution will also not give any Karpenter based approach to enable AZ resiliency. **The feature will be gated under an alpha feature flag to allow for additional feedback from customers, and make subsequent changes in the future.**
+
+### Option  1 (recommended) : Unhealthy condition set cloud provider interface  
+
+```
+type RepairPolicy struct {
+    // Type of unhealthy state that is found on the node
+    Type metav1.ConditionType 
+    // Status condition of when a node is unhealthy
+    Status metav1.ConditionStatus
+    // RepairAfter is the duration the controller will wait
+    // before attempting to terminate nodes that are marked for repair.
+    TolerationDuration time.Duration
+}
+
+type CloudProvider interface {
+  ...
+    // RepairPolicy is for CloudProviders to define a set Unhealthy condition for Karpenter 
+    // to monitor on the node. Customer will need 
+    RepairPolicy() []v1.RepairPolicy
+  ...
+}
+```
+
+The `RepairPolicy` will be a set of condition that the Karpenter controller will watch. The cloud provider can define compatibility with any node diagnostic agent, and track a list of node unhealthy condition types and a duration period to wait until a unhealthy state is considered a terminal: 
+
+1. A diagnostic agent, such as [node problem detector](https://github.com/kubernetes/node-problem-detector), will add a status condition on to a node 
```

Comment:
nit: This is the first time that you are bringing up the term "diagnostic agent". If you are going to use this throughout the doc, I'd recommend bringing up the term earlier in a definitions section